internal/rpc/core/mempool.go

```diff
@@ -117,19 +118,30 @@ func (env *Environment) BroadcastTxCommit(ctx context.Context, tx types.Tx) (*co
 	}
 }
 
-// UnconfirmedTxs gets unconfirmed transactions (maximum ?limit entries)
+// UnconfirmedTxs gets unconfirmed transactions
 // including their number.
 // More: https://docs.tendermint.com/master/rpc/#/Info/unconfirmed_txs
-func (env *Environment) UnconfirmedTxs(ctx context.Context, limitPtr *int) (*coretypes.ResultUnconfirmedTxs, error) {
+func (env *Environment) UnconfirmedTxs(ctx context.Context, pagePtr, perPagePtr *int) (*coretypes.ResultUnconfirmedTxs, error) {
 	// reuse per_page validator
-	limit := env.validatePerPage(limitPtr)
 
-	txs := env.Mempool.ReapMaxTxs(limit)
+	txs := env.Mempool.ReapMaxTxs(-1)
```

Comment:
This is a minor optimisation but say there are 5000 txs in the mempool and we've requested the first page of 100. This function will iterate over the entire mempool unnecessarily. Better would be to do something like:
```
totalCount := env.Mempool.Size()

. . .

txs := env.Mempool.ReapMaxTxs(skipCount+tmmath.MinInt(perPage, totalCount-skipCount)
result := txs[skipCount : ]
```

Reply:
This actually makes a lot of sense